addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/blob/s3/S3BlobStore.java

```diff
@@ -676,6 +678,15 @@ public void writeBlobProperties(BlobUpdateContext blobUpdateContext) throws IOEx
                 logTrace("rnote right: " + status.toString());
                 amazonS3.setObjectLegalHold(request);
             }
+            if (blobUpdateContext.restoreForDuration != null) {
+                Duration duration = blobUpdateContext.restoreForDuration.duration;
+                // round up duration to days
+                int days = (int) duration.plusDays(1).minusSeconds(1).toDays();
+                RestoreObjectRequest request = new RestoreObjectRequest(bucketName, bucketKey);
+                request.setVersionId(versionId);
+                request.setExpirationInDays(days);
+                amazonS3.restoreObjectV2(request);
+            }
```

Comment:
i am wondering if should use/explore the returned object:
` RestoreObjectResult restoreObjectResult = amazonS3.restoreObjectV2(request);`
mainly the use of `com.amazonaws.services.s3.model.RestoreObjectResult#restoreOutputPath`
```
    /**
     * The path in the provided S3 output location where Select results will be restored to
     */
    private String restoreOutputPath;
```

Or we shouldn't care about it and when the content is available  call the  `org.nuxeo.ecm.core.blob.BlobProvider#getStatus` then we can make the call to `org.nuxeo.ecm.core.blob.BlobProvider#getStream`

Reply:
`restoreOutputPath` is about "select" restores, which is something different.
In our case there's really nothing interesting in the result... It's just launching an async restore.